syscalls_darwin.go

```diff
@@ -159,7 +159,16 @@ func (t *tunReadCloser) Write(from []byte) (int, error) {
 	}
 	t.wBuf = t.wBuf[:len(from)+4]
 
-	t.wBuf[3] = 2 // Family: IP (2)
+	// Determine the IP Family for the NULL L2 Header
+	ipVer := from[0] >> 4
```

Comment:
Would you mind doing a check on the length of `from` here? Otherwise if a `nil` value or empty slice is passed in this would panic. Although I'm not exactly sure what we should set `t.wBuf[3]` to in this case. I guess a zero value is fine?